src/Uno.UI.RuntimeTests/MUX/Microsoft_UI_Xaml_Controls/TreeViewTests.cs

```diff
@@ -0,0 +1,716 @@
+			Verify.IsTrue(selectionChangedEventArgs.AddedItems.Contains(1));
+			Verify.AreEqual(0, selectionChangedEventArgs.RemovedItems.Count);
+			selectionChangedEventArgs = null;
+
+			tvi2.IsSelected = true;
+
+			Verify.IsNotNull(selectionChangedEventArgs);
+			Verify.AreEqual(1, selectionChangedEventArgs.AddedItems.Count);
+			Verify.IsTrue(selectionChangedEventArgs.AddedItems.Contains(2));
+			Verify.AreEqual(1, selectionChangedEventArgs.RemovedItems.Count);
+			Verify.IsTrue(selectionChangedEventArgs.RemovedItems.Contains(1));
+			selectionChangedEventArgs = null;
+
+			tvi2.IsSelected = false;
+
+			Verify.IsNotNull(selectionChangedEventArgs);
+			Verify.AreEqual(0, selectionChangedEventArgs.AddedItems.Count);
+			Verify.AreEqual(1, selectionChangedEventArgs.RemovedItems.Count);
+			Verify.IsTrue(selectionChangedEventArgs.RemovedItems.Contains(2));
+			selectionChangedEventArgs = null;
+
+			tvi1.IsSelected = true;
+
+			Verify.IsNotNull(selectionChangedEventArgs);
+			Verify.AreEqual(1, selectionChangedEventArgs.AddedItems.Count);
+			Verify.IsTrue(selectionChangedEventArgs.AddedItems.Contains(1));
+			Verify.AreEqual(0, selectionChangedEventArgs.RemovedItems.Count);
+			selectionChangedEventArgs = null;
+
+			treeView.ItemsSource = null;
+
+			Verify.IsNotNull(selectionChangedEventArgs);
+			Verify.AreEqual(0, selectionChangedEventArgs.AddedItems.Count);
+			Verify.AreEqual(1, selectionChangedEventArgs.RemovedItems.Count);
+			Verify.IsTrue(selectionChangedEventArgs.RemovedItems.Contains(1));
+		}
+
+		[TestMethod]
+		public void TreeViewSelectionChangedMultipleMode()
+		{
+
+			// input data:
+			// - 1
+			//   - 11
+			//   - 12
+			//   - 13
+			// - 2
+			//   - 21
+			// - 3
+			TreeViewSelectionChangedEventArgs selectionChangedEventArgs = null;
+
+			var treeView = new TreeView { SelectionMode = TreeViewSelectionMode.Multiple };
+			treeView.SelectionChanged += (s, e) => selectionChangedEventArgs = e;
+
+			var node1 = new TreeViewNode { Content = "1", IsExpanded = true };
+			var node11 = new TreeViewNode { Content = "11" };
+			var node12 = new TreeViewNode { Content = "12" };
+			var node13 = new TreeViewNode { Content = "13" };
+			node1.Children.Add(node11);
+			node1.Children.Add(node12);
+			node1.Children.Add(node13);
+
+			var node2 = new TreeViewNode { Content = "2", IsExpanded = true };
+			var node21 = new TreeViewNode { Content = "21" };
+			node2.Children.Add(node21);
+
+			var node3 = new TreeViewNode { Content = "3" };
+
+			treeView.RootNodes.Add(node1);
+			treeView.RootNodes.Add(node2);
+			treeView.RootNodes.Add(node3);
+			TestServices.WindowHelper.WindowContent = treeView;
+			treeView.UpdateLayout();
+
+			var tvi1 = (TreeViewItem)treeView.ContainerFromItem(node1);
+			var tvi11 = (TreeViewItem)treeView.ContainerFromItem(node11);
+			var tvi12 = (TreeViewItem)treeView.ContainerFromItem(node12);
+			var tvi13 = (TreeViewItem)treeView.ContainerFromItem(node13);
```

Comment:
Codacy found an issue: [Remove the unused local variable 'tvi13'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5156819)